Sources/Apollo/ApolloStore.swift

```diff
@@ -252,6 +252,23 @@ public final class ApolloStore {
       try self.cache.removeRecord(for: key)
     }
 
+    /// Removes records with keys that match the specified pattern. This method will only
+    /// remove whole records, it does not perform cascading deletes. This means only the
+    /// records with matched keys will be removed, and not any references to them. Key
+    /// matching is case-insensitive.
+    ///
+    /// If you attempt to pass a cache key for a single field, this method will do nothing
```

Comment:
I'd love us to get some terminology to differentiate between object and field keys as I think it's a common source of  confusion. 
Thinking about it, I don't think "field keys" make a lot of sense. It's not something that can be used as a key in a dictionary or a SQL primary key column. We still use them in watchers to get notified when a field changes but I think using the "key" concept there is misleading there. 

What about `CachePath` instead? For the below query:

```graphql
{
  hero {
    name 
    friend {
      email
    }
  }
}
```

We'd have:

* CacheKeys: `hero`, `hero.friends.0`, `hero.friends.1`, etc..
* CachePaths used by watchers: 
  * `hero`, `hero.friends` (do we even need those?)
  * `hero.name`
  * `hero.friends` (in case the list changes, new items moved, added or removed)

This specific API would take a `CacheKey`, making it explicit that it only works at the "object" level.
Thoughts?

Reply:
Yes to the common language, à la [Uniquitous Language from Domain Driven Design](https://martinfowler.com/bliki/UbiquitousLanguage.html).

`CachePath` does seem the more correct and approproiate.

@designatednerd, @AnthonyMDev - any comment/preference?

Reply:
I like cache path. We have too many overloaded and vague terms at apollo. I'm always for being more careful with our terminology.